frontend/src/components/Login/Login.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from "react"
+import basestyle from "../base.module.css"
+import loginstyle from "./login.module.css"
+import axios from "axios"
+import { useNavigate, NavLink } from "react-router-dom"
+const Login = () => {
+  const navigate = useNavigate()
+  const [formErrors, setFormErrors] = useState({})
+  const [user, setUserDetails] = useState({
+    email: "",
+    password: "",
+  })
+  const [error, setError] = useState("")
+
+  const changeHandler = (e) => {
+    const { name, value } = e.target
+    setUserDetails({
+      ...user,
+      [name]: value,
+    })
+  }
+  const validateForm = (values) => {
+    const error = {}
+    const regex = /^[^\s+@]+@[^\s@]+\.[^\s@]{2,}$/i
+    if (!values.email) {
+      error.email = "Email is required"
+    } else if (!regex.test(values.email)) {
+      error.email = "Please enter a valid email address"
+    }
+    if (!values.password) {
+      error.password = "Password is required"
+    }
+    return error
+  }
+
+  const loginHandler = async (e) => {
+    e.preventDefault()
+    setFormErrors(validateForm(user))
+    try {
+      setError("")
+      const res = await axios.post(
+        "https://backend-dun-nine.vercel.app/users/login",
```

Comment:
What if you wanted to check something that is not deployed yet? What is needs to be changed?

Reply:
to check something that is not deployed we can simply run yarn start in the frontend

Reply:
and to check something that is not deployed in the backend as well we can do the same

Reply:
Ha... but you wanted to connect with localhost instead of backedn-dun-nine.vercel.app. Isn't it?